sbin/root-mount

```diff
@@ -149,18 +162,28 @@ if ! lvm vgchange -a y sailfish; then
 	exit 1
 fi
 
-# Check factory reset need.
+# Check factory reset need and read FS resize status.
 if mount -t ext4 $ROOTDEV $1; then
 	factory_reset_if_needed $1
+
+	if test -f "$1/$FS_RESIZED"; then
+		IS_FS_RESIZED=1
+	fi
+
 	umount $1
 fi
 
 # Check and resize root and home in case usage indicates first boot
-check_firstboot_resize
+if ! check_firstboot_resize; then
+	echo "First boot resize failed. Aborting boot." > /dev/kmsg
+	exit 1
```

Comment:
This is quite risky to exit here as that means device does not even try to boot. Even if something fails for some strange reason we should try to continue booting so user has some possibility of recovery of his data in such case.